### ISSUE TITLE
update_saved_status should acquire exclusive lock

### ISSUE THREAD

**Issue by rimms** [jubatus/jubatus#1113]
We should acquire a write-lock.

- https://github.com/jubatus/jubatus/blob/0.9.0/jubatus/server/framework/server_base.cpp#L184
- https://github.com/jubatus/jubatus/blob/0.9.0/jubatus/server/framework/server_base.cpp#L221-L224

**Comment by rimms** [comment]
I tested with the following codes:
https://github.com/rimms/misc/tree/master/save_loop

``jubaclassifier`` often be aborted with the above test. 

Backtrace:
```
(gdb) bt
#0  0x00007f72c22130d5 in __GI_raise (sig=<optimized out>) at ../nptl/sysdeps/unix/sysv/linux/raise.c:64
#1  0x00007f72c221683b in __GI_abort () at abort.c:91
#2  0x00007f72c225061e in __libc_message (do_abort=2, fmt=0x7f72c235ab40 "*** glibc detected *** %s: %s: 0x%s ***\n")
    at ../sysdeps/unix/sysv/linux/libc_fatal.c:201
#3  0x00007f72c225ae16 in malloc_printerr (action=3, str=0x7f72c235ad30 "double free or corruption (fasttop)", ptr=<optimized out>)
    at malloc.c:5047
#4  0x00007f72c2a6eff6 in std::string::assign(std::string const&) () from /usr/lib/x86_64-linux-gnu/libstdc++.so.6
#5  0x00007f72c4bf5aef in jubatus::server::framework::server_base::save (this=0x24c9c00, id=...)
    at ../jubatus/server/framework/server_base.cpp:184
#6  0x000000000041a596 in jubatus::server::classifier_impl::save (this=0x7fffe013f260, id=...)
    at ../jubatus/server/server/classifier_impl.cpp:92

 .. snip ..

(gdb) f 5
#5  0x00007f72c4bf5aef in jubatus::server::framework::server_base::save (this=0x24c9c00, id=...)
    at ../jubatus/server/framework/server_base.cpp:184
184	  update_saved_status(path);
```

**Comment by rimms** [comment]
Fixed via #1115
